webui/react/src/stores/cluster.tsx

```diff
@@ -93,17 +92,16 @@ const clusterStatusText = (
   )}%`;
 };
 
-const updateIfChanged = <T, V extends WritableObservable<T>>(o: V, next: T) =>
-  o.update((prev) => (_.isEqual(prev, next) ? prev : next));
-
 class ClusterStore extends PollingStore {
-  #agents: WritableObservable<Loadable<Agent[]>> = observable(NotLoaded);
-  #resourcePools: WritableObservable<Loadable<ResourcePool[]>> = observable(NotLoaded);
-  #unboundResourcePools: WritableObservable<Loadable<ResourcePool[]>> = observable(NotLoaded);
-  #resourcePoolBindings: WritableObservable<Map<string, number[]>> = observable(Map());
+  #agents = deepObservable<Loadable<Agent[]>>(NotLoaded);
+  #resourcePools = deepObservable<Loadable<ResourcePool[]>>(NotLoaded);
+  #unboundResourcePools = deepObservable<Loadable<ResourcePool[]>>(NotLoaded);
+  #resourcePoolBindings = immutableObservable<Map<string, OrderedSet<number>>>(Map());
 
   public readonly agents = this.#agents.readOnly();
-  public readonly resourcePoolBindings = this.#resourcePoolBindings.readOnly();
+  public readonly resourcePoolBindings = this.#resourcePoolBindings.select((bindings) =>
+    bindings.map((workspaceIds) => workspaceIds.toJS()),
```

Comment:
Any particular reason you didn't want to return an OrderedSet?

Reply:
For this pr, I wanted to make sure the only changes were to the bookkeeping done inside the stores -- external types should remain the same so we know that if bugs occur it's because of the update checking, not the type change.